src/jsx.d.ts

```diff
@@ -1499,7 +1499,78 @@ export namespace JSXInternal {
 		rel?: string | undefined | SignalLike<string | undefined>;
 		required?: boolean | undefined | SignalLike<boolean | undefined>;
 		reversed?: boolean | undefined | SignalLike<boolean | undefined>;
-		role?: string | undefined | SignalLike<string | undefined>;
+		role?:
+			| 'alert'
+			| 'alertdialog'
+			| 'application'
+			| 'article'
+			| 'banner'
+			| 'button'
+			| 'cell'
+			| 'checkbox'
+			| 'columnheader'
+			| 'combobox'
+			| 'complementary'
+			| 'contentinfo'
+			| 'definition'
+			| 'dialog'
+			| 'directory'
+			| 'document'
+			| 'feed'
+			| 'figure'
+			| 'form'
+			| 'grid'
+			| 'gridcell'
+			| 'group'
+			| 'heading'
+			| 'img'
+			| 'link'
+			| 'list'
+			| 'listbox'
+			| 'listitem'
+			| 'log'
+			| 'main'
+			| 'marquee'
+			| 'math'
+			| 'menu'
+			| 'menubar'
+			| 'menuitem'
+			| 'menuitemcheckbox'
+			| 'menuitemradio'
+			| 'navigation'
+			| 'none'
+			| 'note'
+			| 'option'
+			| 'presentation'
+			| 'progressbar'
+			| 'radio'
+			| 'radiogroup'
+			| 'region'
+			| 'row'
+			| 'rowgroup'
+			| 'rowheader'
+			| 'scrollbar'
+			| 'search'
+			| 'searchbox'
+			| 'separator'
+			| 'slider'
+			| 'spinbutton'
+			| 'status'
+			| 'switch'
+			| 'tab'
+			| 'table'
+			| 'tablist'
+			| 'tabpanel'
+			| 'term'
+			| 'textbox'
+			| 'timer'
+			| 'toolbar'
+			| 'tooltip'
+			| 'tree'
+			| 'treegrid'
+			| 'treeitem'
+			| undefined
+			| SignalLike<string | undefined>;
```

Comment:
Wow, thanks for doing this!

We'll also want to make sure signals users are able to benefit from these much better types:

```suggestion
			| SignalLike<
				| 'alert'
				| 'alertdialog'
				| 'application'
				| 'article'
				| 'banner'
				| 'button'
				| 'cell'
				| 'checkbox'
				| 'columnheader'
				| 'combobox'
				| 'complementary'
				| 'contentinfo'
				| 'definition'
				| 'dialog'
				| 'directory'
				| 'document'
				| 'feed'
				| 'figure'
				| 'form'
				| 'grid'
				| 'gridcell'
				| 'group'
				| 'heading'
				| 'img'
				| 'link'
				| 'list'
				| 'listbox'
				| 'listitem'
				| 'log'
				| 'main'
				| 'marquee'
				| 'math'
				| 'menu'
				| 'menubar'
				| 'menuitem'
				| 'menuitemcheckbox'
				| 'menuitemradio'
				| 'navigation'
				| 'none'
				| 'note'
				| 'option'
				| 'presentation'
				| 'progressbar'
				| 'radio'
				| 'radiogroup'
				| 'region'
				| 'row'
				| 'rowgroup'
				| 'rowheader'
				| 'scrollbar'
				| 'search'
				| 'searchbox'
				| 'separator'
				| 'slider'
				| 'spinbutton'
				| 'status'
				| 'switch'
				| 'tab'
				| 'table'
				| 'tablist'
				| 'tabpanel'
				| 'term'
				| 'textbox'
				| 'timer'
				| 'toolbar'
				| 'tooltip'
				| 'tree'
				| 'treegrid'
				| 'treeitem'
				| undefined
			>;
```